spec/webhook_spec.rb

```diff
@@ -0,0 +1,115 @@
+        and_return(sample_response)
+
+      @mailgun.webhooks.find(@sample[:id])
+    end
+  end
+
+  describe "add a webhook" do
+    context "using the default webhook url" do
+      it "should make a POST request with correct params to add a webhook" do
+        sample_response = "{\"message\":\"Webhook has been created\",\"webhook\":{\"url\":\"http://postbin.heroku.com/860bcd65\"}}"
+        webhooks_url = @mailgun.webhooks.send(:webhook_url)
+
+        Mailgun.should_receive(:submit).
+          with(:post, webhooks_url, {:id => @sample[:id], :url => @sample[:url]}).
+          and_return(sample_response)
+
+        @mailgun.webhooks.create(@sample[:id])
+      end
+    end
+    context "overwriting the default webhook url" do
+      it "should make a POST request with correct params to add a webhook" do
+        sample_response = "{\"message\":\"Webhook has been created\",\"webhook\":{\"url\":\"http://postbin.heroku.com/860bcd65\"}}"
+        webhooks_url = @mailgun.webhooks.send(:webhook_url)
+        overwritten_url = 'http://mailgun.net/webhook'
+
+        Mailgun.should_receive(:submit).
+          with(:post, webhooks_url, {:id => @sample[:id], :url => overwritten_url}).
+          and_return(sample_response)
+
+        @mailgun.webhooks.create(@sample[:id], overwritten_url)
+      end
+    end
+  end
+
+  describe "update a webhook" do
+    context "using the default webhook url" do
+      it "should make a POST request with correct params to add a webhook" do
+        sample_response = "{\"message\":\"Webhook has been updated\",\"webhook\":{\"url\":\"http://postbin.heroku.com/860bcd65\"}}"
+        webhooks_url = @mailgun.webhooks.send(:webhook_url, @sample[:id])
+
+        Mailgun.should_receive(:submit).
+          with(:put, webhooks_url, {:url => @sample[:url]}).
+          and_return(sample_response)
+
+        @mailgun.webhooks.update(@sample[:id])
+      end
+    end
+    context "overwriting the default webhook url" do
+      it "should make a POST request with correct params to add a webhook" do
+        sample_response = "{\"message\":\"Webhook has been updated\",\"webhook\":{\"url\":\"http://postbin.heroku.com/860bcd65\"}}"
+        webhooks_url = @mailgun.webhooks.send(:webhook_url, @sample[:id])
+        overwritten_url = 'http://mailgun.net/webhook'
+
+        Mailgun.should_receive(:submit).
+          with(:put, webhooks_url, {:url => overwritten_url}).
+          and_return(sample_response)
+
+        @mailgun.webhooks.update(@sample[:id], overwritten_url)
+      end
+    end
+  end
+
+  describe "delete a webhook" do
+    it "should make a DELETE request with correct params" do
+      sample_response = "{\"message\":\"Webhook has been deleted\",\"webhook\":{\"url\":\"http://postbin.heroku.com/860bcd65\"}}"
+      webhooks_url = @mailgun.webhooks.send(:webhook_url, @sample[:id])
+
+      Mailgun.should_receive(:submit).
+        with(:delete, webhooks_url).
+        and_return(sample_response)
```

Comment:
normal style is to indent the second line of method chaining. This isn't really our repo though, so this is a non-blocking comment.